tinygrad/engine/realize.py

```diff
@@ -11,9 +11,17 @@ def __init__(self, fxn):
     super().__init__()
   def __call__(self, rawbufs:List[Buffer], var_vals:Dict[Variable, int], wait=False, jit=False): self.fxn(*rawbufs)
 
+def maybe_zerocopy(si: ScheduleItem) -> bool:
+  if not (all([len(x) == 1 for x in [si.outputs, si.inputs, si.ast]]) and si.outputs[0].device.startswith("DISK")): return False
+  ast, out, inp = si.ast[0], si.outputs[0], si.inputs[0]
+  top_src = ast.src[0].src[0] if ast.src[0].op is UnaryOps.CAST and ast.src[0].arg[1] else ast.src[0]
+  if not (top_src.op is BufferOps.LOAD and top_src.arg.st.consecutive and ast.arg.st.contiguous) or hasattr(out, "_buf"): return False
+  out.base, out.offset = (inp.base or inp), inp.offset+(top_src.arg.st.views[0].offset*inp.dtype.itemsize)
```

Comment:
Oh, I see, allocate won't trigger because of this...

Reply:
I don't think it is possible to do it cleanly. I've tried and some of the problems are:
- toposort in schedule doesn't understand that realizing one buffer can realize its zero copy childs and teaching it do understand it isn't very clean 
- there are different ways to trigger zero copy (contiguous, bitcast, assign) and they can be nested (eg assign after bitcast). This makes it hard to trigger on lazybuffer level but all of this is already taken care of in scheduling so it's a simple pattern match with bitcast skip afterwards (see maybe_zerocopy function before i removed it)

I think the most clean way to do it is to move maybe_zerocopy between schedule_one which takes care of contiguous/bitcast/assign+combinations of them already and the place where multioutput will merge ScheduleItems.

Reply:
But "then that buffer will become "allocated" when the parent is" is a very good idea and i've implemented it so that zero copy can be triggered before realization of base

Reply:
Done.

@geohot can you enable benchmars/additional tests/whatever this is for this pr? tiny13 is busy again (and seems to be +-always busy recently) with some wandb stuff